### PR TITLE
Security: Lots of backend changes

### PR DESCRIPTION
**Pull request by Jmartimus**
Here is what changed:

-Added hashing and salting of new passwords on the back end when signing up using bcrypt.
-Login method now signs a JWT and returns an access token.
-When Login method occurs in the FE, it sets the access token in session storage.
-I also added some helpful functions in the API methods file for future work. (Sign out, automatic push to other page, what to include in every axios request to be authenticated.).
-I added guards to all transaction methods on the backend.  They now require authentication to access.
-The transaction methods check for a user_id to make sure that they don't return someone else's data.
-Refactored expenses and incomes into transactions.

### CODE REVIEW

File: client/src/API/AuthMethods.ts [hunk 1]
```diff
@@ -19,13 +18,32 @@ export const signUp = async (userInfo: UserInfoData) => {
...
+    },
+      //  This needs to be included in all of our other methods with axios.
+      // { withCredentials: true, headers: { 'Authorization': `Bearer ${sessionStorage.getItem('authToken')}` } }
+    );
+    sessionStorage.setItem('authToken', response.data.accessToken);
+    if (response) {
+      // write code to automatically push user to next page.
+      // history.push('/');
+    }
+  } catch (e) {
```
Comment (bycho91) [inline client/src/API/AuthMethods.ts:33]: do we no longer need to type the error or should I still research this later?

**Review by bycho91 — COMMENTED** [review]
I don't understand everything that's going on here, but I didn't find anything wrong after going through.

I just have to make note of the controller route change to 'transactions' once these changes are merged!

**Review by bycho91 — APPROVED** [review]
😃

**Review by bretpeters3n — APPROVED** [review]
It all looks legit. I'll merge this for you/

**Pull request merged**